src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheck.java

```diff
@@ -65,16 +65,16 @@
  * Default value is {@code false}.
  * </li>
  * <li>
- * Property {@code allowSamelineSingleParameterlessAnnotation} - Allow single parameterless
+ * Property {@code allowSamelineParameterizedAnnotation} - Allow one and only parameterized
  * annotation to be located on the same line as target element.
  * Type is {@code boolean}.
- * Default value is {@code true}.
+ * Default value is {@code false}.
  * </li>
  * <li>
- * Property {@code allowSamelineParameterizedAnnotation} - Allow one and only parameterized
+ * Property {@code allowSamelineSingleParameterlessAnnotation} - Allow single parameterless
  * annotation to be located on the same line as target element.
  * Type is {@code boolean}.
- * Default value is {@code false}.
+ * Default value is {@code true}.
```

Comment:
The reason why the metadata file and the javadoc are changed is because of the order of the properties. Properties didn't have any order until now(or do they?) so it's impossible to match their order. The order now follows the order of the setters in the module class.

Reply:
XDocs Order: https://github.com/checkstyle/checkstyle/blob/54649c930bfd0fc09f59fb5afdd8c6df82c680a5/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1551

XdocsJavaDocsTest took XDoc as it is  (order and all) and made sure it was in the correct format as the Javadoc.

JavadocMetadataScraper just took what was in Javadoc and placed it into meta file.

It seems to me, javadoc should be ordered just like xdoc, which is alphabetical order by TreeSet.

Reply:
Looks like your issue is at:
https://github.com/checkstyle/checkstyle/pull/13407/files#diff-582038065de9c242c07931401cfa262fcfd745638bf7c42736dba5793a3294cdR401

You switch over from the TreeSet to a LinkedHashSet. You shouldn't use hash sorting as different VMs can possibly create different hashes. Even if we were guaranteed hashes, we want alpha ordering here.

Reply:
> I recommend creating a new issue for this and this being done first. It won't be easy to review if there is a bunch of property changes like this.

I second this.

Reply:
Can we implement alphabet ordering now, but do not migrate numerous checks for now. So we can do:
 1)migration to macroses in smaller batches 

2)make massive updates for ordering first and then separate PRs wave to use macroses.

We will see later on what is less on effort.